harmonica/_equivalent_sources/cartesian.py

```diff
@@ -178,6 +178,11 @@ def __init__(
                 f"Found invalid 'depth' value equal to '{depth}'. "
                 "It should be 'default' or a numeric value."
             )
+        if depth == 0:
```

Comment:
Perhaps check if depth <= 0.

Because depth cannot be smaller than zero

Reply:
Hey @VascoSch92, 

From the perspective of the equivalent source, you can have sources above the observation point. The only problem arises when you have them in the same position as the observation points. 

Cheers